### ISSUE TITLE
Remove NoDeleteMixin

### ISSUE THREAD

**Issue by gutard** [GeotrekCE/Geotrek-admin#2136]

**Comment by gutard** [comment]
Some consequences:
* When a topology has no more path it will be deleted definitively
* The caching mecanism relies on last modification time of deleted objects so it is broken

**Comment by camillemonchicourt** [comment]
Yes removing the boolean field "supprime" would simplify several things and was finally not really used as a temporary trash.

Yes to remove a topology when deleting its last path. I think there is a message before confirmation mentioning related objects that will be removed too.

**Comment by gutard** [comment]
La suppression du NoDeleteMixin pose une autre difficulté. Lors de nombreux tests, on crée des topologies sans aggrégation. Du coup, la geom est NULL mais un trigger passe automatiquement la topologie en deleted=TRUE ce qui est autorise la geom à être NULL dans ce cas précis. Sans NoDeleteMixin, la topologie est réellement supprimée et le test ne peut pas se poursuivre. Ou alors il faudrait autoriser la geom a être NULL tout le temps ce qui n'est pas souhaitable.

=> Décision de reporter la suppression du NoDeleteMixin à plus tard, après les changements sur la segmentation dynamique.

**Comment by camillemonchicourt** [comment]
Or use django-safedelete ? 
https://github.com/GeotrekCE/Geotrek-admin/issues/813
